lib/errors.js

```diff
@@ -15,7 +16,7 @@ function error(code, data, err) {
     if (data) {
         // ** If the second argument is an error, then reset the arguments.
         if (isError(data))
-            return error(code, null, data);
+            return error(code, null, data, err);
```

Comment:
I'm a bit confused here, if `data` is already an error, what would we receive as `err`?